src/Shouldly.Tests/ShouldBeLessThanOrEqualTo/CustomObjectScenario.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Shouldly.Tests.Strings;
+using Shouldly.Tests.TestHelpers;
+using Xunit;
+
+namespace Shouldly.Tests.ShouldBeLessThanOrEqualTo
+{
+    public class CustomObjectScenario
+    {
+        [Fact]
+        public void CustomObjectScenarioShouldFail()
+        {
+            var customA = new Custom { Val = 2 };
+            var customB = new Custom { Val = 1 };
+            var comparer = new CustomComparer<Custom>();
+            Verify.ShouldFail(() =>
+customA.ShouldBeLessThanOrEqualTo(customB, comparer, "Some additional context"),
+
+errorWithSource:
+@"customA
+    should be less than or equal to
+Shouldly.Tests.TestHelpers.Custom (000000)
+    but was
+Shouldly.Tests.TestHelpers.Custom (000000)
+
```

Comment:
Yeah, this is not really possible without either overriding ToString or us trying to print out the values of members which could be too noisy.